lib/ravelin/ato_login.rb

```diff
@@ -0,0 +1,13 @@
+module Ravelin
+  class AtoLogin < RavelinObject
+    attr_accessor :login,
+      :device,
+      :location
+
+    def login=(l)
+      @login = Ravelin::Login.new(l)
+    end
+
+    attr_required :login
```

Comment:
Stylistically, it would be more conventional to put this kind of macro towards the top of the file, before 'normal' methods.